src/transformers/generation/configuration_utils.py

```diff
@@ -386,125 +597,116 @@ class GenerationConfig(PushToHubMixin):
+    cache_config: Optional[Dict] = None
+    if cache_implementation is not None and cache_implementation in CACHE_CONFIG_MAPPING:
+        cache_config_class = CACHE_CONFIG_MAPPING[cache_implementation]
+        if isinstance(cache_config, dict):
+            cache_config = cache_config_class.from_dict(cache_config)
+    return_legacy_cache: Optional[bool] = None
+
+    # Parameters for manipulation of the model output logits
+    temperature: Optional[confloat(ge=0.0, le=2.0)] = 1.0
+    top_k: Optional[conint(ge=0)] = 50
+    top_p: Optional[confloat(ge=0.0, le=1.0)] = 1.0
+    min_p: Optional[confloat(ge=0.0, le=1.0)] = None
+    typical_p: Optional[confloat(ge=0.0, le=1.0)] = 1.0
+    epsilon_cutoff: Optional[confloat(ge=0.0, le=1.0)] = 0.0
+    eta_cutoff: Optional[confloat(ge=0.0, le=1.0)] = 0.0
+    diversity_penalty: Optional[confloat(ge=0.0)] = 0.0
+    repetition_penalty: Optional[confloat(ge=1.0)] = 1.0
+    encoder_repetition_penalty: Optional[confloat(ge=1.0)] = 1.0
+    length_penalty: float = 1.0
+    no_repeat_ngram_size: Optional[conint(ge=0)] = 0
+    bad_words_ids: Optional[List[List[int]]] = None
+    force_words_ids: Optional[List[Union[List[int], List[List[int]]]]] = None
+    renormalize_logits: Optional[bool] = False
+    constraints: Optional[List[Dict]] = None
+    forced_bos_token_id: Optional[conint(ge=0)] = None
+    forced_eos_token_id: Optional[Union[conint(ge=0), List[conint(ge=0)]]] = None
+    remove_invalid_values: Optional[bool] = False
+    exponential_decay_length_penalty: Optional[Union[tuple, list]] = None
+    suppress_tokens: Optional[Union[int, List[int]]] = None
+    begin_suppress_tokens: Optional[List[int]] = None
+    forced_decoder_ids: Optional[List[List[int]]] = None
+    sequence_bias: Optional[Union[List[Union[List[int], float]], List[List[Union[List[int], float]]]]] = None
+    token_healing: Optional[bool] = False
+    guidance_scale: Optional[confloat(ge=1.0)] = None
+    low_memory: Optional[bool] = None
+    watermarking_config: Optional[Union[dict, "WatermarkingConfig", "SynthIDTextWatermarkingConfig"]] = None
+    if watermarking_config is None:
+        watermarking_config = None
+    elif isinstance(watermarking_config, BaseWatermarkingConfig):
+        watermarking_config = watermarking_config
+    else:
+        watermarking_config = WatermarkingConfig.from_dict(watermarking_config)
+
+    # Parameters that define the output variables of `generate`
+    num_return_sequences: Optional[conint(ge=1)] = 1
+    output_attentions: Optional[bool] = False
+    output_hidden_states: Optional[bool] = False
+    output_scores: Optional[bool] = False
+    output_logits: Optional[bool] = None
+    return_dict_in_generate: Optional[bool] = False
+
+    # Special tokens that can be used at generation time
+    pad_token_id: Optional[int] = None
+    bos_token_id: Optional[conint(ge=0)] = None
+    eos_token_id: Optional[Union[conint(ge=0), List[conint(ge=0)]]] = None
```

Comment:
```suggestion
    eos_token_id: Optional[Union[int, List[int]]] = None
```
We might want to set eos to a negative value to ensure this criteria is never hit